server/src/main/java/org/opensearch/index/shard/StoreRecovery.java

```diff
@@ -439,7 +444,7 @@ private ActionListener<Boolean> recoveryListener(IndexShard indexShard, ActionLi
         });
     }
 
-    private void recoverFromRemoteStore(IndexShard indexShard) throws IndexShardRecoveryException {
+    private void recoverFromRemoteStore(IndexShard indexShard, Repository repository) throws IndexShardRecoveryException {
```

Comment:
Use `Optional` here as well?

Reply:
`Optional<Repository>` as a parameter? But optional generally used in the return type, right?